contracts/modules/common/BaseTransfer.sol

```diff
@@ -150,4 +161,31 @@ abstract contract BaseTransfer is BaseModule {
             usedAllowance,
             _data);
     }
+
+    /**
+    * @notice Helper method to wrap ETH into WETH, approve a certain amount of WETH and call an external contract.
+    * The address that spends the WETH and the address that is called with _data can be different.
+    * @param _wallet The target wallet.
+    * @param _spender The spender address.
+    * @param _amount The amount of tokens to transfer.
+    * @param _contract The contract address.
+    * @param _data The method data.
+    */
+    function doApproveWethAndCallContract(
+        address _wallet,
+        address _spender,
+        uint256 _amount,
+        address _contract,
+        bytes memory _data
+    )
+        internal
+    {
+        uint256 wethBalance = ERC20(wethToken).balanceOf(_wallet);
```

Comment:
I guess having this function abstracted here has its benefits but the downside is having to check the token balance for the wallet twice, firstly here then again in `doApproveTokenAndCallContract`. I think I slightly lean towards your current option though.

Reply:
Yes, I think getting rid of `doApproveWethAndCallContract` and having the ETH wrapping logic added to `doApproveTokenAndCallContract` would be reasonable but I kind of like to keep this "special case" separate.